src/evalfunc.c

```diff
@@ -7540,17 +7541,18 @@ f_id(typval_T *argvars, typval_T *rettv)
 	case VAR_BLOB:
 	    // Assume pointer value in typval_T vval union at common location.
 	    if (argvars[0].vval.v_object != NULL)
-		vim_snprintf((char*)numbuf, sizeof(numbuf), "%p",
-					    (void *)argvars[0].vval.v_object);
-	    else
-		numbuf[0] = NUL;
-	    break;
+	    {
+		vimlong_T v = (vimlong_T)(void *)argvars[0].vval.v_object;
+		for(; p < numbuf + sizeof(void *) * 2; ++p, v >>= 4)
```

Comment:
Substitute `<< 1` for `* 2` (and parenthesise)?

Reply:
Except for `p` they're all constants; the compiler figures it out.

(using `<<` would require parens)